src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -1,14 +1,52 @@
 const { readFile } = require('fs/promises');
 const { join } = require('path');
 const { cwd } = require('process');
+const { getPackument } = require('query-registry');
+
+const dependenciesGlobal = {};
 
 async function getDependencies() {
-  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+  if (Object.keys(dependenciesGlobal).length === 0) {
+    const dependencyList = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+
+    // The order of the alternatives is important!
+    // The regex engine will favor the first pattern on an alternation
+    // even if the other alternatives are subpatterns
+    for (const dependencyName of dependencyList.split(/\r\n|\n|\r/).filter((line) => line !== '')) {
```

Comment:
I think this is enough.
```suggestion
    for (const dependencyName of dependencyList.split(/\r\n?/).filter((line) => !!line)) {
```

Reply:
I believe that macOS uses `\r` for line ending. Is that the case?

Reply:
`\r\n?` includes '\r` for unix system and `\r\n` for windows.